xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/XMLUtils.java

```diff
@@ -63,6 +64,59 @@
  */
 public final class XMLUtils
 {
+    /**
+     * A simple error logging helper to suppress expected error messages.
+     * While extracting text excerpts via the {@link #extractXML} method
+     * the extraction throws expected exceptions eg. in case a text node
+     * is too long. The default ErrorListener writes these errors to the console.
+     * However as these errors are expected under normal conditions
+     * and are either handled in the {@link #extractXML} method
+     * or passed up in the chain of commands it is ok to log these
+     * at a much lower level, where users normally never see them.
```

Comment:
I don't understand fully this comment. Since you log the error it'll be seen in the logs by users. It seems this is just replacing the XML parser logging with an xwiki logging. Right? Basically, I don't see how this suppresses error messages. Thanks!

Reply:
I've now read more and this does 2 things
- replaces the default ErrorListener which logs to System.err
- swallows all logs for severities error and below, and only logs fatal errors

Reply:
> and are either handled in the {@link #extractXML} method

I don't think that's the case since the default error listener captures all exceptions and logs them, so they're never handled by extractXML AFAIK. Maybe I misunderstood what you meant?

Reply:
The default error listener captures all exceptions, but _does not handle_ them.

Instead the code inside the `Transformer` looks like this:

        } catch (RuntimeException e) {
            if (_errorListener != null) postErrorToListener(e.getMessage());
            throw new TransformerException(e);
        }

so any errors thrown by the transformation process still have to be handled by the calling side (just wrapped into a `TransformerException`) and are not swallowed by the error listener.

The calling side on xwiki-commons then seem to have the information to deal with these (wrapped) exceptions:

        } catch (Throwable t) {
            if (handler != null && handler.isFinished()) {
                return handler.getResult();
            } else {
                throw new RuntimeException("Failed to extract XML", t);
            }
        }

I am not sure if this is the right way to handle the exceptions but this is how the code is, independent on what I did in the pull request. Seemingly the way to handle this is first to check if the ``handler`` is finished - in this case the exception has been used for flow control, i.e. to signal that the rest of the document does not need to be processed (there seem to be no other way to stop the processing). Otherwise it is rethrown to be dealt with by the calling side (which probably shows an error message in the UI, but as this method can be called from many places I am not sure what it does).

In no case an exception is swallowed by the error listener. It is swallowed by the ``extractXML`` when that method guesses that the actual job to extract a text snippet has been successfully achieved.